sdk-base/src/main/java/com/mapbox/maps/extension/style/StyleContract.kt

```diff
@@ -135,4 +140,16 @@ interface StyleContract {
      */
     fun bindTo(delegate: StyleInterface)
   }
+
+  /**
+   * Define the interfaces for the model extension.
+   */
+  fun interface StyleModelExtension {
```

Comment:
IMO model here is pretty vast... did you consider `Style3DModelExtension`? I understand that will effect all the code in this PR but better think about it now IMO

Reply:
Yeah, just noticed that we have `enum class ModelType` and values there basically show that it's not only 3D